core/src/main/java/org/opensearch/sql/datasource/model/DataSourceMetadata.java

```diff
@@ -11,26 +11,29 @@
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
-import lombok.NoArgsConstructor;
 import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
 import org.opensearch.sql.datasource.DataSourceService;
 
 @Getter
 @Setter
 @AllArgsConstructor
-@NoArgsConstructor
 @EqualsAndHashCode
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class DataSourceMetadata {
 
   @JsonProperty private String name;
 
+  @JsonProperty private String description;
```

Comment:
Does this annotation provide default getter/setter? Do not see code for that

Reply:
There is getter setter annotation above class name. JsonProperty is usually not required unless we have some customization in serialization like changing the name of the field to something other than field name/ignoring in serialization.